omod/src/test/java/org/openmrs/module/fhir2/providers/EncounterFhirResourceProviderWebTest.java

```diff
@@ -12,22 +12,29 @@
 import static org.hamcrest.MatcherAssert.assertThat;
```

Comment:
This also needs some tests verifying complex behaviour. I.e., what should happen if I request
`/Encounter?subject.given=Harold&location.address-postalcode=90210,90177`?

Reply:
@ibacher What should happen in this case? 

```java
@Test
	public void shouldGetEncountersBySubjectGivenNameAndLocationPostalCode() throws Exception {
		verifyUri("/Encounter?subject.given=Harold&?location.address-postalcode=90210,90177");

		verify(encounterService).searchForEncounters(isNull(), locationCaptor.capture(), isNull(), subjectCaptor.capture());
		assertThat(subjectCaptor.getValue(), notNullValue());
		assertThat(locationCaptor.getValue(), notNullValue());
		assertThat(subjectCaptor.getValue().getChain(), equalTo("given"));
		assertThat(subjectCaptor.getValue().getValue(), equalTo(PATIENT_GIVEN_NAME));
		assertThat(locationCaptor.getValue().getChain(), equalTo("address-postalcode"));
		assertThat(locationCaptor.getValue().getValue(), equalTo(ENCOUNTER_POSTALCODE));
	}
```

The error failed with error message
```

java.lang.AssertionError: 
Expected: response with HTTP status indicating request was handled successfully
     but: response with status code <400> with message "Invalid request: The FHIR endpoint on this server does not know how to handle GET operation[Encounter] with parameters [[?location.address-postalcode, subject.given]]"
```

Reply:
@ibacher Its working if I use only one parameter in location

Reply:
Oh yeah, fair enough, we'd have to change it to a `ReferenceOrListParam`, which probably isn't worth it for this PR.

Reply:
@ibacher I think I have fixed everything else. So is it ready for being merged